pkg/daemon/update.go

```diff
@@ -890,8 +909,9 @@ func verifyUserFields(pwdUser ign3types.PasswdUser) error {
 	if tempUser.Name == constants.CoreUserName && len(tempUser.SSHAuthorizedKeys) >= 1 {
 		tempUser.Name = ""
 		tempUser.SSHAuthorizedKeys = nil
+		tempUser.PasswordHash = nil
 		if !reflect.DeepEqual(emptyUser, tempUser) {
-			return fmt.Errorf("ignition passwd user section contains unsupported changes: non-sshKey changes")
+			return fmt.Errorf("ignition passwd user section contains unsupported changes. Supported changes includes Password Hash and SSH key")
 		}
 		glog.Info("SSH Keys reconcilable")
 	} else {
```

Comment:
error messaging in else condition needs to be updated based on how we fix comment  https://github.com/openshift/machine-config-operator/pull/3539/files#r1106978031 .
Note that with passwordhash support, user  can specify any of these:
1. ssh key for user core
2. password hash for user core
3. ssh key and password hash for user core